integrations/instagram/src/index.ts

```diff
@@ -64,34 +97,50 @@ const integration = new bp.Integration({
       const token = query['hub.verify_token']
       const challenge = query['hub.challenge']
 
-      if (mode === 'subscribe' && token === ctx.configuration.verifyToken) {
-        if (!challenge) {
-          return {
-            status: 400,
+      if (mode === 'subscribe') {
+        if (token === ctx.configuration.verifyToken) {
+          if (!challenge) {
+            logger.forBot().warn('Returning HTTP 400 as no challenge parameter was received in query string of request')
+            return {
+              status: 400,
+            }
           }
-        }
 
-        return {
-          body: typeof challenge === 'string' ? challenge : '',
+          return {
+            body: typeof challenge === 'string' ? challenge : '',
+          }
+        } else {
+          logger
+            .forBot()
+            .warn("Returning HTTP 403 as the Instagram token doesn't match the one in the bot configuration")
+          return {
+            status: 403,
+          }
         }
       } else {
+        logger.forBot().warn(`Returning HTTP 400 as the '${mode}' mode received in the query string isn't supported`)
         return {
-          status: 403,
+          status: 400,
         }
       }
     }
 
     if (!req.body) {
-      console.warn('Handler received an empty body')
+      logger.forBot().warn('Handler received an empty body, so the message was ignored')
       return
     }
 
-    const data = JSON.parse(req.body) as InstagramPayload
+    try {
+      const data = JSON.parse(req.body) as InstagramPayload
 
-    for (const { messaging } of data.entry) {
-      for (const message of messaging) {
-        await handleMessage(message, { client, ctx, logger })
+      for (const { messaging } of data.entry) {
+        for (const message of messaging) {
+          await handleMessage(message, { client, ctx, logger })
+        }
       }
+    } catch (e: any) {
+      logger.forBot().error('Error while handling request:', e)
+      logger.forBot().debug('Request body received:', req.body)
```

Comment:
This is probably not necessary? As the request body is already logged above in line 91?

Reply:
I feel like it makes sense to make debugging easier